client/extensions/hello-dolly/state/selectors.js

```diff
@@ -0,0 +1,11 @@
+/**
+ * Internal Dependencies
+ */
+import lyrics from './lyrics';
+
+export function getCurrentLyric( reduxState ) {
+	const state = reduxState.helloDolly;
+
+	return lyrics[ state.lyricIndex ];
```

Comment:
another note, especially given that this is example code for others, is a use of dependency injection that I use in order to make testing much easier. notice how we cannot test or call `getCurrentLyric()` without the hard coupling to the lyrics array…

```js
export const getCurrentLyric = lyrics => state => {
	…
}

export default getCurrentLyric( lyrics );
```

in this code the function takes the array of lyrics as its first curried argument and returns the new and _more real_ selector as a result. now if we want to test the function we can use simple data mocks…

```js
import { getCurrentLyric } from '…';

describe( '#getCurrentLyric', () => {
	const threeSongs = [ 'one', 'two', 'three' ];
	const dollyState = state => ( { helloDolly: state } )

	it( 'should return the first item if no state index exists', () => {
		expect( getCurrentLyric( threeSongs )( dollyState( {} ) ).to.equal( 'one' );
	} );

	it( 'should wrap-around in the list if given out-of-bounds indices', () => {
		expect( getCurrentLyric( threeSongs )( dollyState( { lyricIndex: 2 } ).to.equal( 'three' );
		expect( getCurrentLyric( threeSongs )( dollyState( { lyricIndex: 3 } ).to.equal( 'one' );
		expect( getCurrentLyric( threeSongs )( dollyState( { lyricIndex: 4 } ).to.equal( 'two' );
	} );
} );
```

and now we have a couple tests which expressly test the _behavior_ of the function